pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -126,30 +130,24 @@ func (*IIDAttestorPlugin) GetPluginInfo(context.Context, *spi.GetPluginInfoReque
 	return &spi.GetPluginInfoResponse{}, nil
 }
 
-func (p *IIDAttestorPlugin) getConfig() (*IIDAttestorConfig, error) {
-	p.mtx.RLock()
-	defer p.mtx.RUnlock()
-
-	if p.config == nil {
-		return nil, errors.New("not configured")
+// endpointFromLegacyConfig extracts the endpoint from legacy configuration values
+// This code is transitional, here until these deprecated configs are removed
+func endpointFromLegacyConfig(docURL, sigURL string) (string, error) {
+	docSuffix := "/dynamic/" + docPath
+	if !strings.HasSuffix(docURL, docSuffix) {
+		return "", fmt.Errorf("IID URL '%s' doesn't end in expected suffix %s", docURL, docSuffix)
 	}
-	return p.config, nil
-}
+	docEndpoint := strings.TrimSuffix(docURL, docSuffix)
 
-func httpGetBytes(url string) ([]byte, error) {
-	resp, err := http.Get(url) //nolint: gosec // URL is provided via explicit configuration
-	if err != nil {
-		return nil, err
+	sigSuffix := "/dynamic/" + sigPath
+	if !strings.HasSuffix(sigURL, sigSuffix) {
+		return "", fmt.Errorf("IID signature URL '%s' doesn't end in expected suffix %s", sigURL, sigSuffix)
 	}
-	defer resp.Body.Close()
+	sigEndpoint := strings.TrimSuffix(sigURL, sigSuffix)
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	if docEndpoint != sigEndpoint {
+		return "", fmt.Errorf("IID URL and Signature URL had different endpoints: %s != %s", docEndpoint, sigEndpoint)
```

Comment:
This is source #2 of incompatibility:  If you use two different endpoints it doesn't work -- the SDK only takes a single one.